app/src/components/governance/PageProposal.vue

```diff
@@ -271,119 +279,83 @@ export default {
     },
   },
   apollo: {
-    proposals: {
-      query() {
-        /* istanbul ignore next */
-        return gql`
-          query proposals($networkId: String!) {
-            proposals(networkId: $networkId) {
-              id
-              status
-            }
-          }
-        `
-      },
-      variables() {
-        /* istanbul ignore next */
-        return {
-          networkId: this.network,
-        }
-      },
-      update(data) {
-        if (!data.proposals) {
-          return []
-        }
-        /* istanbul ignore next */
-        if (
-          data.proposals.find(
-            (proposal) => proposal.id === parseInt(this.proposalId, 10)
-          )
-        ) {
-          this.found = true
-        }
-        /* istanbul ignore next */
-        return data.proposals
-      },
-    },
     proposal: {
+      /* istanbul ignore next */
       query() {
-        /* istanbul ignore next */
         return ProposalItem(this.network)
       },
+      /* istanbul ignore next */
       update(data) {
-        /* istanbul ignore next */
         this.loaded = true
-        /* istanbul ignore next */
+        if (data.proposal) this.found = true
         return data.proposal || {}
       },
+      /* istanbul ignore next */
       variables() {
-        /* istanbul ignore next */
         return {
-          id: +this.proposalId,
+          id: this.proposalId,
         }
       },
-      skip() {
-        /* istanbul ignore next */
-        return !this.found
-      },
+      /* istanbul ignore next */
       result(data) {
-        /* istanbul ignore next */
+        if (data.proposal) this.found = true
         this.error = data.error
       },
     },
     parameters: {
+      /* istanbul ignore next */
       query() {
-        /* istanbul ignore next */
         return GovernanceParameters(this.network)
       },
+      /* istanbul ignore next */
       update(data) {
-        /* istanbul ignore next */
-        return data.governanceParameters
+        return data.governanceParameters || {}
       },
+      /* istanbul ignore next */
       skip() {
-        /* istanbul ignore next */
-        return !this.found
+        // only Tendermint networks have this network-wide "governance parameters" logic
```

Comment:
wouldn't it be simpler to return an empty object or a message than to have network specific logic in the FE like this?

Reply:
true, but we will always have protocol specific logic in the FE

Reply:
sure sure - but this feels like the wrong place to me. maybe up in the template...?

Reply:
template? What do you mean? I think this is a good place. Explains the skip